WordPress/openverse
# Update jest packages

## Fixes



Fixes #4047 by @obulat

## Description


This PR updates jest and jest-related packages (babel-, testing library).

To test that this allows `script setup` in components, I converted `VImageCell` to `script setup`. This also showed that there is an unused function, `getImageForeignUrl`, which is removed in this PR.

I also commented out `mockCreateApiService`'s`afterEach` because it causes the following error: `Error: Hooks cannot be defined inside tests. Hook of type "afterEach" is nested within "should save the token into the process and inject into the context".` This did not change the test results. I don't completely understand the setup there, would removing `afterEach` be okay, @sarayourfriend ?

## Testing Instructions

The unit tests in the CI should pass.
The image search results (both images and all results search views) should work correctly and not show additional warnings.

## Checklist



[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
